### ISSUE TITLE
Perform missing data analysis

### ISSUE THREAD

**Issue by mattmills49** [mattmills49/Kaggle_Comps#5]
I want to know how frequent certain variables are missing but also how frequently observations are missing the same variables? Is there systemic missing values?

**Comment by mattmills49** [comment]
Interesting pattern between where variables are missing values

![test](https://raw.githubusercontent.com/mattmills49/Kaggle_Comps/master/Santander/graphs/missing_around-1.png)

Note I excluded any observations where no values are missing, so this is just looking at observations where at least one thing is missing.

**Comment by apryor6** [comment]
Are the observations ordered chronologically? If so, looks like there was an abrupt change in the structure of data entries

edit: Looked at the code and answered my own question - it's sorted chronologically. Might be worth separating data before and after the date of the change, or at least cleaning them separately.

**Comment by mattmills49** [comment]
Yup you were right, the data was sorted by month. Here is a look at which months had the highest rates of missing values:

![](https://github.com/mattmills49/Kaggle_Comps/raw/master/Santander/graphs/all_missing_values-1.png)

I still think the other graph is important, we just need to isolate out the months effect. The other graph is really about the relationship between variables. When one variable is missing does that mean others are, and it definitely looks like there is. This is important because combined with the month to month missingness the data clearly aren't missing at random and we have to be smart about replacing the values. Granted, we don't have to replace them, some algorithms can handle missing values and for categorical variables we can just treat the missing values as a new level.
